### PR TITLE
fix(www): fixes the containers used in all guidelines page by converting them to theme-ui

### PR DESCRIPTION
**Pull request by abhishekjakhar**
# PR

## Description

The guidelines pages are not working on mobile screen sizes, basically images are not showing up and scrolling feels like infinite scroll, the next section comes after scrolling a lot.

This is happening because all of the guideline pages have not been converted to theme-ui and none of the media query rule is working in entire set of components used inside guidelines(color, logo, typography, design tokens).

This PR fixes all of the containers used in Guidelines pages and make all of the Guideline pages work on responsive. This is first step to fix all the guidelines pages. We also need to fix each of the page individually by converting them to theme-ui, I am working on that simultaneously, I will raise PR for each of the page individually after this gets merged.
I did not include all the changes in single PR, because it would make this PR to be very big.

Converted the <Layout/> component of Guidelines to theme-ui

##### Logo page
logo-before

##### Colors page
colors-before

##### Typography page
typo-before

##### Design tokens page
design-tokens-before

##### Logo page
logo-after

##### Colors page
The Colors page will get fixed after converting the components used here to theme-ui, it cannot be fixed by just fixing the containers

##### Typography page
typo-after

##### Design tokens page
design-tokens-after

## Related Issues

Fixes https://github.com/gatsbyjs/gatsby/issues/24218

### CODE REVIEW

**Comment by fk** [comment]
<3 Thank you Abhishek! 🎉
 I don't think I'll manage to review today anymore, but I will tomorrow morning (~12 hours).

File: www/src/components/guidelines/containers.js [hunk 1]
```diff
@@ -1,39 +1,65 @@
...
-import propTypes from "@styled-system/prop-types"
+import { mediaQueries } from "gatsby-design-tokens/dist/theme-gatsbyjs-org"
 
-import { Box } from "./system"
-
-export const Container = styled(Box)()
-
-Container.defaultProps = {
+const container = {
+  boxSizing: `border-box`,
```
Comment (fk) [inline www/src/components/guidelines/containers.js:6]: Curious why you explicitly added this rule — did you not have `border-box` here without it? It should be set for all elements by default via the `*` selector: 
image

File: www/src/components/guidelines/containers.js [hunk 2]
```diff
@@ -1,39 +1,65 @@
...
+  <section
+    sx={{
+      ...rest,
+      ...container,
+      py: 4,
+      [mediaQueries.sm]: {
+        py: 5,
+      },
+      [mediaQueries.md]: {
+        py: 8,
```
Comment (fk) [inline www/src/components/guidelines/containers.js:36]: Did you try the "responsive array" `theme-ui` syntax here, and it didn't work? If you want to not set a value for an individual breakpoint, you can do e. g. `px: [ 2, null, null, 3 ]` IIRC.

**Comment by abhishekjakhar** [comment]
> 🎉 @abhishekjakhar! 🙏
> 
> AFAICS, we have a minor regressions for larger screens — "Typography"'s "Scale" content container, and "Font Sizes" in https://www.gatsbyjs.org/guidelines/design-tokens/ both were set to be "full bleed" and extend all the way to the edge of the viewport —
> 
> image
> 
> — but _way more importantly_, this fixes the mobile layout, and gets us started on removing the confusing direct usage of `styled-system`. Thank you so much for that, Abhishek!
> 
> Let's get this in, we can always improve stuff later on!

@fk I have already already kept 4 PR ready for all the 4 pages, as soon as this gets merged, I will raise 4 more PR's for color, logo, typo and design-tokens page respectively, they will fix everything on both mobile and desktop. Also, they will convert each of the individual page to theme-ui. Just waiting for this to get merged hahaha. 
Thank you very much.

File: www/src/components/guidelines/containers.js (hunk 1 above)
Comment (abhishekjakhar) [reply www/src/components/guidelines/containers.js:6]: Earlier we were using `import { Box } from "./system"`, and this Box which we were importing was explicitly adding this rule,I thought there must be some reason behind it, so I kept it untouched. I will remove it now as it is getting set by `*` selector.
Thank you very much

**Comment by fk** [comment]
> I will raise 4 more PR's for color, logo, typo and design-tokens page respectively, they will fix everything on both mobile and desktop. Also, they will convert each of the individual page to theme-ui. 

🙏 🤗 🎉

File: www/src/components/guidelines/containers.js (hunk 1 above)
Comment (fk) [reply www/src/components/guidelines/containers.js:6]: Ahh OK! 👍 

Most probably a leftover from the prototype then, which probably didn't use https://www.gatsbyjs.org/packages/gatsby-plugin-typography/, which normalizes `box-sizing` for us in the case of .org.

**Pull request merged**